src/common/hooks/useWarnIfUnsavedChanges.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from "react";
```

Comment:
does this only work if you refresh the page completely ? (full refresh and not click on a link ?)

Reply:
no it works as long as the user tries to navigate away

Reply:
What's the magic? How to tell to not display a message ? It works because the component is unmounted ?

Reply:
Yea so it listens on the [beforeUnload](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event) event